storage/range_tree_test.go

```diff
@@ -46,3 +68,443 @@ func TestIsRed(t *testing.T) {
 		}
 	}
 }
+
+// TestFlip ensures that flips function correctly.
+func TestFlip(t *testing.T) {
```

Comment:
Given all these tests in this file (which are better as unittests than the ones in client_range_tree_test), as well as the randomized test just added, my feeling is we should drop the super verbose tests in client_range_tree_test which verify specific tree structures. I think we've overtested now between the two and those super verbose ones would be my choice to eliminate as they're going to be hard for anyone else to maintain.